examples/codegen/codegen.cpp

```diff
@@ -40,6 +40,24 @@ std::vector<ur_adapter_handle_t> get_adapters() {
     return adapters;
 }
 
+std::vector<ur_adapter_handle_t>
+get_supported_adapters(std::vector<ur_adapter_handle_t> &adapters) {
+    std::vector<ur_adapter_handle_t> supported_adapters;
+    size_t size = 0;
+    for (auto adapter : adapters) {
+        ur_adapter_backend_t backend;
+        ur_check(urAdapterGetInfo(adapter, UR_ADAPTER_INFO_BACKEND,
+                                  sizeof(ur_adapter_backend_t), &backend,
+                                  nullptr));
+
+        if (backend == UR_ADAPTER_BACKEND_LEVEL_ZERO) {
```

Comment:
I filtered that on only l0 as I think the other adapter that would accept spirv is opencl and that is still not merged in intel/llvm so we could add it later.